Automattic/jetpack
# Boost: Update feature status to use svelte-data-sync-client

Related to #29414

## Proposed changes:

* Used wp-js-data-sync library for module statuses
* Updated compatibility between old and new boost versions
* Refactored compatibility between Jetpack and boost

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Toggle modules and make sure they work correctly
* Make sure score refreshes when toggling modules
* Make sure toggling modules hides/shows correct UI elements under each module
* Enabling critical CSS and cloud CSS should trigger critical CSS gen when necessary
